src/bson.rs

```diff
@@ -386,6 +386,7 @@ impl Bson {
                     "$symbol" => (v.to_owned())
                 }
             }
+            Bson::Document(ref v) => v.to_owned(),
```

Comment:
I don't think this is a correct implementation. You must at least convert each values to its extended form, instead of cloning the `Document` itself.

Specification: https://docs.mongodb.com/manual/reference/mongodb-extended-json/